saleor/payment/gateways/braintree/__init__.py

```diff
@@ -136,128 +123,118 @@ def get_client_token(**connection_params: Dict) -> str:
     return client_token
 
 
-def authorize(
-        payment: Payment,
-        payment_token: str,
-        **connection_params: Dict) -> Tuple[Transaction, str]:
+def authorize(payment_information: Dict, **connection_params: Dict) -> Dict:
     gateway = get_braintree_gateway(**connection_params)
+
     try:
         result = gateway.transaction.sale({
-            'amount': str(payment.total),
-            'payment_method_nonce': payment_token,
+            'amount': str(payment_information['amount']),
+            'payment_method_nonce': payment_information['token'],
             'options': {
                 'submit_for_settlement': CONFIRM_MANUALLY,
                 'three_d_secure': {
                     'required': THREE_D_SECURE_REQUIRED}},
-            **get_customer_data(payment)})
+            **get_customer_data(payment_information)})
     except braintree_sdk.exceptions.NotFoundError:
-        return transaction_and_incorrect_token_error(
-            payment, kind=TransactionKind.AUTH, token=payment_token)
+        raise BraintreeException
```

Comment:
Docs specify that it can be also an exception class. 

Essentially it doesn't matter whether it's custom or not. Can leave it as it, or remove that block and not catch anything and just return a generic response.. Catching was added by the original author, so I thought to keep it.